### PR TITLE
Adding common error code check

### PR DESCRIPTION
**Pull request by ovinis**
This PR contains error check for all common error codes as described in [REST API Error codes](https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_concepts_errors.htm)

### CODE REVIEW

File: tabcmd/commands/commands.py [hunk 1]
```diff
@@ -55,7 +55,27 @@ def get_data_source_id(server, datasource_name):
...
-            logger.error(message, exception)
+            Commands.check_common_error_codes(logger, exception)
         else:
             logger.error(message)
         sys.exit(1)
+
+    @staticmethod
+    def check_common_error_codes(logger, error):
+        if error.code.find("404") == 0:
+            logger.debug("Not Found: Resource cannot not be located")
```
Comment (jacalata) [inline tabcmd/commands/commands.py:66]: what's the benefit of having both lines? Shouldn't the error line show up in the debug log too?

Comment (jacalata) [reply tabcmd/commands/commands.py:66]: Might also be worth printing the complete code, if this is the 6 digit one?

**Review by jacalata — APPROVED** [review]
So if I read the diff right, the only error code changes are in session and commands, everything else is just formatting. yes?

**Comment by ovinis** [comment]
> So if I read the diff right, the only error code changes are in session and commands, everything else is just formatting. yes?

Yes

File: tabcmd/commands/commands.py (hunk 1 above)
Comment (ovinis) [reply tabcmd/commands/commands.py:66]: Thanks, Fixed. I thought for some reason errors were printed to the console and debug strings went to the log.

**Pull request merged**